### PR TITLE
iox-#743 Add architecture article

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] ~~Tests follow the [best practice for testing][testing]~~
1. [x] ~~Changelog updated [in the unreleased section][changelog] including API breaking changes~~
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer
* Update `.puml`s for `v2.0`
* Add architecture article

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #743 (partly)

### CODE REVIEW

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:9]: The links are broken, please fix them and use `../website/images/iceoryx_software_layers_v2_0_0.svg`

Please update this for all links.

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:17]: Link broken.

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:24]: Link broken.

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:64]: Link broken.

**Review by elfenpiff — CHANGES_REQUESTED** [review]

File: doc/website/concepts/architecture.md
Comment (elBoberido) [reply doc/website/concepts/architecture.md:9]: @mossmaurice would `../images/iceoryx_software_layers_v2_0_0.svg` work? This is different than the hyperlinks since the rendered markdown file looks broken

Comment (mossmaurice) [reply doc/website/concepts/architecture.md:9]: If we don't ever change the position of the images and they'll always be in this folder, I'm ok with relative links for images. Adapted.

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md:17]: Fixed.

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md:24]: See my [comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1203#discussion_r818093610). If we use a relative link here, we would need to adapt it, if this file is moved. If we let `mkdocs-autolinks-plugin` do the magic, we don't ever have to care about this link again unless the file name changes.

I think most users access iceoryx.io via the website and don't read it via the GitHub web GUI reader. If you want I can dig out the click rates for both the iceoryx.io and the related markdown files on GitHub, so we can compare them.

Comment (MatthiasKillat) [reply doc/website/concepts/architecture.md:24]: I personally cannot say what I prefer here. But I have to admit that relative links (or absolute ones) become unmaintainable fast.

Depends on our goals I think, but I do think we should strive to keep the maintenance effort for documentation low (wrt. moving files etc.). If a file is moved we would have to track all files linking it otherwise, which is cumbersome.

Just my 2 cents, others are certainly more knowledgeable than myself in these matters and hence I leave this to them. I just would like it to be as easy as possible to add (technical)  documentation.

File: doc/website/concepts/architecture.md
Comment (MatthiasKillat) [inline doc/website/concepts/architecture.md:62]:
```suggestion
The introspection client can be used for live debugging and provides information of the current system like memory usage and established connections.
```

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md]: Ok, I changed the links relatives ones. I still think we'll (again) end up having maintenance problems, but want to unblock this PR. And continue the discussion after the `v2.0` release.

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md]: Fixed.

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md:35]: Re-phrased.

**Review by MatthiasKillat — APPROVED** [review]

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:24]: When I follow up this link I do not get a detailed description I only come to some simple stub document. Shouldn't we adjust the link to this document: https://github.com/eclipse-iceoryx/iceoryx/blob/ada657a42e8a039e685f9807b11c01fee378678c/iceoryx_hoofs/README.md

File: doc/website/concepts/architecture.md
Comment (elfenpiff) [inline doc/website/concepts/architecture.md:35]: Actually, capro is short for CAnonical PROtocol and implemented originally this pattern here: https://en.wikipedia.org/wiki/Canonical_protocol_pattern

Comment (mossmaurice) [reply doc/website/concepts/architecture.md:35]: Good catch, thanks! Replaced the link.

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md]: Ok, to unblock this discussion. Let's just use the full http link for now..

**Review by elfenpiff — APPROVED** [review]

File: doc/website/concepts/architecture.md
Comment (MatthiasKillat) [inline doc/website/concepts/architecture.md:65]: Do we want to use the relative link here?

**Review by MatthiasKillat — APPROVED** [review]

File: doc/website/concepts/architecture.md
Comment (mossmaurice) [reply doc/website/concepts/architecture.md:65]: I'll fix that on #1211

**Pull request merged**
